benchmark/common.js

```diff
@@ -12,12 +12,50 @@ exports.createBenchmark = function(fn, configs, options) {
 function Benchmark(fn, configs, options) {
   // Use the file name as the name of the benchmark
   this.name = require.main.filename.slice(__dirname.length + 1);
-  // Parse job-specific configuration from the command line arguments
-  const parsed_args = this._parseArgs(process.argv.slice(2), configs);
-  this.options = parsed_args.cli;
-  this.extra_options = parsed_args.extra;
-  // The configuration list as a queue of jobs
-  this.queue = this._queue(this.options);
+  this.queue = [];
+  const byGroup = (options && options.byGroup) || false;
+
+  const enqueueConfigsInFirstGroup = (configs) => {
+    const firstGroupKey = Object.keys(configs)[0];
+    const configsInFirstGroup = configs[firstGroupKey];
+    const parsedArgs =
+      this._parseArgs(process.argv.slice(2), configsInFirstGroup);
+    this.options = parsedArgs.cli;
+    this.extra_options = parsedArgs.extra;
+    this.queue = this._queue(this.options);
+  };
+
+  const enqueueConfigsInGroups = (configs) => {
+    for (const groupKey of Object.keys(configs)) {
+      const config = configs[groupKey];
+      const parsedArgs = this._parseArgs(process.argv.slice(2), config);
+      this.options = parsedArgs.cli;
+      this.extra_options = parsedArgs.extra;
+      // The configuration list as a queue of jobs by merging the existing
+      // items with the new items to return a new array
+      this.queue = [ ...this.queue, ...this._queue(this.options) ];
+    }
+  };
+
+  const enqueueConfigs = (configs) => {
+    // Parse job-specific configuration from the command line arguments
+    const parsedArgs = this._parseArgs(process.argv.slice(2), configs);
+    this.options = parsedArgs.cli;
+    this.extra_options = parsedArgs.extra;
+    // The configuration list as a queue of jobs
+    this.queue = this._queue(this.options);
+  };
+
+  if (byGroup) {
+    if (process.env.NODEJS_BENCHMARK_BYPASS_GROUP) {
+      enqueueConfigsInFirstGroup(configs);
+    } else {
+      enqueueConfigsInGroups(configs);
+    }
+  } else {
+    enqueueConfigs(configs);
+  }
```

Comment:
I think this can be simplified to
```js
const parseConfig = (config) => {
      const parsedArgs = this._parseArgs(process.argv.slice(2), config);
      this.options = parsedArgs.cli;
      this.extra_options = parsedArgs.extra;
      this.queue = [ ...this.queue, ...this._queue(this.options) ];
}

if (byGroup) {
  let usedConfigs = Object.values(configs);
  if (process.env.NODEJS_BENCHMARK_BYPASS_GROUP)
    usedConfigs = usedConfigs[0];
  usedConfigs.forEach(parseConfig);
} else {
  parseConfig(configs);
}
```